protocol/contracts/templegold/SpiceAuction.sol

```diff
@@ -0,0 +1,494 @@
+
+    /**
+     * @notice Recover auction tokens for epoch with zero bids
+     * @param epochId Epoch Id
+     * @param to Recipient
+     */
+    function recoverAuctionTokenForZeroBidAuction(uint256 epochId, address to) external override onlyDAOExecutor {
+        if (to == address(0)) { revert CommonEventsAndErrors.InvalidAddress(); }
+        // has to be valid epoch
+        if (epochId > _currentEpochId) { revert InvalidEpoch(); }
+        if (epochsWithoutBidsRecovered[epochId]) { revert AlreadyRecovered(); }
+        // epoch has to be ended
+        EpochInfo storage epochInfo = epochs[epochId];
+        if (!epochInfo.hasEnded()) { revert AuctionActive(); }
+        // bid token amount for epoch has to be 0
+        if (epochInfo.totalBidTokenAmount > 0) { revert InvalidOperation(); }
+
+        SpiceAuctionConfig storage config = auctionConfigs[epochId];
+        (, address auctionToken) = _getBidAndAuctionTokens(config);
+        epochsWithoutBidsRecovered[epochId] = true;
+        uint256 amount = epochInfo.totalAuctionTokenAmount;
+        _totalAuctionTokenAllocation[auctionToken] -= amount;
+
+        emit CommonEventsAndErrors.TokenRecovered(to, auctionToken, amount);
+        IERC20(auctionToken).safeTransfer(to, amount);
+    }
+
+    /// @notice withdraw ETH used for layer zero sends
+    function withdrawEth(address payable _to, uint256 _amount) external override onlyOperatorOrDaoExecutor {
+        if (_to == address(0)) { revert CommonEventsAndErrors.InvalidAddress(); }
+        if (_amount == 0) { revert CommonEventsAndErrors.ExpectedNonZero(); }
+        uint256 _balance = address(this).balance;
+        if (_balance < _amount) { revert CommonEventsAndErrors.InsufficientBalance(address(0), _amount, _balance); }
+
+        (bool success, ) = _to.call{ value: _amount }("");
+        if (!success) { revert WithdrawFailed(_amount); }
+    }
+
+    /**
+     * @notice Burn redeemd TGLD and notify circulating supply
+     * @param epochId Epoch Id
+     * @param useContractEth If to use contract eth for layerzero send
+     */
+    function burnAndNotify(uint256 epochId, bool useContractEth) external payable override nonReentrant {
+        if (redeemedEpochs[epochId]) { revert CommonEventsAndErrors.InvalidParam(); }
+        EpochInfo storage epochInfo = epochs[epochId];
+        if (epochInfo.startTime == 0) { revert InvalidEpoch(); }
+        if (!epochInfo.hasEnded()) { revert AuctionActive(); }
+
+        SpiceAuctionConfig storage _config = auctionConfigs[epochId];
+        (address bidToken,) = _getBidAndAuctionTokens(_config);
+        if (bidToken != templeGold) { revert CommonEventsAndErrors.InvalidParam(); }
+        uint256 amount = epochInfo.totalBidTokenAmount;
+        if (amount == 0) { revert CommonEventsAndErrors.ExpectedNonZero(); }
+        
+        emit RedeemedTempleGoldBurned(epochId, amount);
+        redeemedEpochs[epochId] = true;
+        _burnAndNotify(amount, _config.recipient, useContractEth);
```

Comment:
This function can revert if not enough tokens are held in `recipient`. The `recipient` can move funds and DoS this function. I guess this is a multisig, so not a big deal. 

Perhaps, if `bidToken`==`templeGold`, the bid tokens should be kept in this contract instead of sent to `recipient`.

Reply:
I think it's safe to assume admin/multisig is not a bad actor and that the contract is given enough approval by recipient